airbyte-integrations/connectors/source-elasticsearch/source_elasticsearch/source.py

```diff
@@ -0,0 +1,179 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from airbyte_cdk.sources import Source
+from elasticsearch import Elasticsearch
+
+
+class UnsupportedDataTypeException(Exception):
+    pass
+
+
+class SourceElasticsearch(Source):
+    system_indices = [
+        ".kibana_1",
+        ".opendistro_security",
+    ]
+
+    es_to_json_type_mapping = {
+        # TODO: support objects, arrays, etc
+        # ES data types: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html
+        # JSON data types: http://json-schema.org/understanding-json-schema/reference/type.html
```

Comment:
Without objects and array merge this become very complicate @olivermeyer. They are very simple and native data types in ElasticSearch. Please make sure those types are supported.